### PR TITLE
FileMerger: added option to concatenate files in RT

### PR DESCRIPTION
**Pull request by hendrikweisser**
Added the option group "rt_concat" to FileMerger. It allows to merge input files (mzML, featureXML, consensusXML) in a way that creates non-overlapping RT regions, by adjusting the RT ranges of the inputs before merging. The RT transformations can be written to output files, which would allow splitting of the merged file and retransforming to the original RT scales (but there is no tool for this yet).

Solves #1500. Also contains a fix for #1499 and some code improvements.

### CODE REVIEW

File: src/topp/FileMerger.cpp [hunk 1]
```diff
@@ -204,6 +257,12 @@ class TOPPFileMerger :
             it->setMetaValue("file_origin", DataValue(file_list[i]));
           }
         }
+
+        if (rt_gap_ > 0.0) // concatenate in RT
+        {
+          adjustRetentionTimes_(map, trafo_out[i], i == 0);
```
Comment (timosachsenberg) [inline src/topp/FileMerger.cpp]: i == 0. Seems to be your thing right now ;)
Would a descriptive bool hurt a lot?

File: src/topp/FileMerger.cpp [hunk 2]
```diff
@@ -309,65 +359,60 @@ class TOPPFileMerger :
...
             writeLog_(String("Warning: No valid retention time for output scan '") + rt_auto + "' from file '" + filename + "'");
           }
 
-          out.addSpectrum(*it2);
-          out.getSpectra().back().setRT(rt_final);
-          out.getSpectra().back().setNativeID(native_id);
-
-          if (user_ms_level)
+          spec_it->setRT(rt_final);
+          spec_it->setNativeID("spectrum=" + String(native_id));
```
Comment (timosachsenberg) [inline src/topp/FileMerger.cpp]: at some point we might think about whether we can preserve the native id instead of setting it to a new one - but that code is not related to your commit

**Comment by timosachsenberg** [comment]
+1

File: src/topp/FileMerger.cpp (hunk 2 above)
Comment (hendrikweisser) [inline src/topp/FileMerger.cpp]: Yes, see my comment here: https://github.com/OpenMS/OpenMS/issues/1500#issuecomment-125604468

File: src/topp/FileMerger.cpp (hunk 1 above)
Comment (hendrikweisser) [inline src/topp/FileMerger.cpp]: I think it's sufficiently clear. "i == 0" was already used to test for the first file elsewhere in the FileMerger code (now line 417).

That said, I can add a bool variable if you really want.

**Comment by lars20070** [comment]
@hendrikweisser Could #1577 be related? We now see FileMerger crashes.
